src/components/project/project.service.ts

```diff
@@ -839,7 +839,11 @@ export class ProjectService {
   async getRootDirectory(
     projectId: string,
     session: ISession
-  ): Promise<Directory> {
+  ): Promise<Directory | null> {
+    if (!session.userId || session.userId === this.config.anonUser.id) {
```

Comment:
This may not be the only reason that the user doesn't have access. Shouldn't this check be done with a database call?

Reply:
Okay, let me check

Reply:
@CarsonF I added `SecuredDirectory` type and changed the resolve field type.
And handling no permission, I changed `match` to `optionalMatch` (https://github.com/SeedCompany/cord-api-v3/pull/1070/files#diff-8d8f4e73db837875a146ee377742270bR846) for getting `Directory` associated with that project. So it will return `undefined` if there's no permission and just returned `null` value for a directory with `canRead: false, canEdit: false`